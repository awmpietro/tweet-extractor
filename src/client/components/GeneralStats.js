import React from 'react';
import { Table, CardTitle } from 'reactstrap';
import PropTypes from 'prop-types';

const GeneralStats = ({ stats }) => {
  return (
    <>
      <CardTitle>
        <h4>General User Stats</h4>
      </CardTitle>
      <Table bordered striped>
        <thead>
          <tr>
            <th>Name</th>
            <th>Quantity</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>Profile Use Background Image</td>
            <td>{stats.profile_use_background_image}</td>
          </tr>
          <tr>
            <td>Has Geo Enabled</td>
            <td>{stats.geo_enabled}</td>
          </tr>
          <tr>
            <td>Is Verified</td>
            <td>{stats.verified}</td>
          </tr>
          <tr>
            <td>Has Extended Profile</td>
            <td>{stats.has_extended_profile}</td>
          </tr>
          <tr>
            <td>Is Default Profile</td>
            <td>{stats.default_profile}</td>
          </tr>
          <tr>
            <td>Has Default Profile Image</td>
            <td>{stats.default_profile_image}</td>
          </tr>
          <tr>
            <td>Notifications Enabled</td>
            <td>{stats.notifications}</td>
          </tr>
        </tbody>
      </Table>
    </>
  );
};

GeneralStats.propTypes = {
  stats: PropTypes.object,
};

export default GeneralStats;
