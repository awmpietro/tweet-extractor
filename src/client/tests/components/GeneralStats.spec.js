import React from 'react';
import GeneralStats from '../../components/GeneralStats';
import renderer from 'react-test-renderer';

describe('GeneralStats', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<GeneralStats stats={{}} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});