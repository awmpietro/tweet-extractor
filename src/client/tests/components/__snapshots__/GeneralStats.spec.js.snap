// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeneralStats snapshot renders 1`] = `
Array [
  <div
    className="card-title"
  >
    <h4>
      General User Stats
    </h4>
  </div>,
  <table
    className="table table-bordered table-striped"
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Quantity
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Profile Use Background Image
        </td>
        <td />
      </tr>
      <tr>
        <td>
          Has Geo Enabled
        </td>
        <td />
      </tr>
      <tr>
        <td>
          Is Verified
        </td>
        <td />
      </tr>
      <tr>
        <td>
          Has Extended Profile
        </td>
        <td />
      </tr>
      <tr>
        <td>
          Is Default Profile
        </td>
        <td />
      </tr>
      <tr>
        <td>
          Has Default Profile Image
        </td>
        <td />
      </tr>
      <tr>
        <td>
          Notifications Enabled
        </td>
        <td />
      </tr>
    </tbody>
  </table>,
]
`;
